bot/exts/moderation/silence.py

```diff
@@ -92,88 +112,222 @@ async def _async_init(self) -> None:
+        await channel.set_permissions(role, overwrite=overwrite)
         await self.previous_overwrites.set(channel.id, json.dumps(prev_overwrites))
 
         return True
 
-    async def _schedule_unsilence(self, ctx: Context, duration: Optional[int]) -> None:
+    @staticmethod
+    async def _get_afk_channel(guild: Guild) -> VoiceChannel:
+        """Get a guild's AFK channel, or create one if it does not exist."""
+        afk_channel = guild.afk_channel
+
+        if afk_channel is None:
+            overwrites = {
+                guild.default_role: PermissionOverwrite(speak=False, connect=False, view_channel=False)
+            }
+            afk_channel = await guild.create_voice_channel("mute-temp", overwrites=overwrites)
+            log.info(f"Failed to get afk-channel, created temporary channel #{afk_channel} ({afk_channel.id})")
+
+        return afk_channel
+
+    async def _kick_voice_members(self, channel: VoiceChannel) -> None:
+        """Remove all non-staff members from a voice channel."""
+        log.debug(f"Removing all non staff members from #{channel.name} ({channel.id}).")
+
+        for member in channel.members:
+            if self._helper_role not in member.roles:
+                try:
+                    await member.move_to(None, reason="Kicking member from voice channel.")
+                    log.debug(f"Kicked {member.name} from voice channel.")
+                except Exception as e:
+                    log.debug(f"Failed to move {member.name}. Reason: {e}")
+                    continue
+
+        log.debug("Removed all members.")
+
+    async def _force_voice_sync(self, channel: VoiceChannel) -> None:
+        """
+        Move all non-staff members from `channel` to a temporary channel and back to force toggle role mute.
+
+        Permission modification has to happen before this function.
+        """
+        # Obtain temporary channel
+        delete_channel = channel.guild.afk_channel is None
+        afk_channel = await self._get_afk_channel(channel.guild)
+
+        try:
+            # Move all members to temporary channel and back
+            for member in channel.members:
+                # Skip staff
+                if self._helper_role in member.roles:
+                    continue
+
+                try:
+                    await member.move_to(afk_channel, reason="Muting VC member.")
+                    log.debug(f"Moved {member.name} to afk channel.")
+
+                    await member.move_to(channel, reason="Muting VC member.")
+                    log.debug(f"Moved {member.name} to original voice channel.")
+                except Exception as e:
+                    log.debug(f"Failed to move {member.name}. Reason: {e}")
+                    continue
+
+        finally:
+            # Delete VC channel if it was created.
+            if delete_channel:
+                await afk_channel.delete(reason="Deleting temp mute channel.")
```

Comment:
Just a small something, but I always feel like such aliases should be banned from the surface of earth, even more if they appear in the audit log, 
```suggestion
                await afk_channel.delete(reason="Deleting temporary mute channel.")
```